xarray/core/variable.py

```diff
@@ -993,6 +993,32 @@ def chunk(self, chunks=None, name=None, lock=False):
 
         return type(self)(self.dims, data, self._attrs, self._encoding, fastpath=True)
 
+    def _as_sparse(self, sparse_format=_default, fill_value=dtypes.NA):
+        """
+        use sparse-array as backend.
+        """
+        import sparse
+
+        # TODO  what to do if dask-backended?
+        if fill_value is dtypes.NA:
+            dtype, fill_value = dtypes.maybe_promote(self.dtype)
+        else:
+            dtype = self.dtype
```

Comment:
I think it would be a nice idea to use result_type here, in case the fill_value and array have different types.
```suggestion
            dtype = dtypes.result_type(self.dtype, fill_value)
```